scripts/flake-me.sh

```diff
@@ -0,0 +1,72 @@
+if [ "$EUID" -ne 0 ]; then
+  echo "Script is not running as root. Attempting to escalate privileges with sudo..."
+  exec sudo "$0" "$@"
+fi
+
+echo
+echo
+echo "This is a destructive operation, we will be replacing your existing NixOS system configuration"
+echo "with one that uses flakes, based on the productionised DogeboxOS flake."
+echo "Continue (y/n)"
+
+read -n 1 -s confirm
+
+if [ "$confirm" != "y" ]; then
+  echo "Aborting..."
+  exit 1
+fi
+
+echo
+echo
+echo "We're going to move your existing non-flake configuration from /etc/nixos to /etc/nixos-dev"
+echo "This means that your current system changes, if any, should still be applied."
+echo "And that if you want to modify things, change /etc/nixos-dev, as /etc/nixos will be overriden on rebuilds."
+echo "Continue? (y/n)"
+
+read -n 1 -s confirm
+
+if [ "$confirm" != "y" ]; then
+  echo "Aborting..."
+  exit 1
+fi
+
+sudo cp -r /etc/nixos /etc/nixos-dev
+
+echo
+echo
+echo "Copied /etc/nixos to /etc/nixos-dev"
+
+echo
+echo
+echo "Do you need a grub bootloader installed?"
+echo "If you're running in Orbstack or another fancy VM manager, you probably want to answer NO."
+echo "If you're in a fully-fledged VM manager (eg. VirtualBox, VMWare, Proxmox etc), you'll want to answer YES."
+echo "Install bootloader? (y/n)"
+
+read -n 1 -s confirm
+
+if [ "$confirm" = "y" ]; then
+  flake="aarch64-bootloader"
```

Comment:
Do we need to have this script determine whether to `aarch64` or `x86_64`?

Reply:
Oops, yep.